docs/testing/unit-testing.md

```diff
@@ -0,0 +1,413 @@
+- The `OfficeMockObject` constructor does *not* add a `run` method to the mock `Word` object, so it must be added explicitly in the seed object.
+- The `OfficeMockObject` constructor does *not* add all of the Word enum classes to the mock `Word` object, so the `InsertLocation.end` value that is referenced in the add-in method must be added explicitly in the seed object.
+- Because the Office JavaScript APIs library isn't loaded in the node process, the `Word` object that is referenced in the add-in code must be declared and initialized.
+
+```javascript
+const OfficeAddinMock = require("office-addin-mock");
+const myWordAddinFeature = require("../my-word-add-in-feature");
+
+// Create the seed mock object.
+const mockData = {
+  context: {
+    document: {
+      body: {
+        paragraph: {
+          font: {},
+        },
+        // Mock the Body.insertParagraph method.
+        insertParagraph: function (paragraphText, insertLocation) {
+          this.paragraph.text = paragraphText;
+          this.paragraph.insertLocation = insertLocation;
+          return this.paragraph;
+        },
+      },
+    },
+  },
+  // Mock the Word.InsertLocation enum.
+  InsertLocation: {
+    end: "end",
+  },
+  // Mock the Word.run method.
+  run: async function(callback) {
+    await callback(this.context);
+  },
+};
+
+// Create the final mock object from the seed object.
+const wordMock = new OfficeAddinMock.OfficeMockObject(mockData);
+
+// Define and initialize the Word object that is called in the insertBlueParagraph function.
+global.Word = wordMock;
+
+/* Code that calls the test framework goes below this line. */
+
+// Jest test set
+describe("Insert blue paragraph at end tests", () => {
+
+  test("color of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();  
+    expect(wordMock.context.document.body.paragraph.font.color).toBe("blue");
+  });
+
+  test("text of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();
+    expect(wordMock.context.document.body.paragraph.text).toBe("Hello World");
+  });
+})
+```
+
+## Adding mock objects, properties, and methods dynamically when testing
+
+In some scenarios, efficient testing requires that mock objects be created or modified at runtime; that is, while the tests are running. The following are examples:
+
+- The function that is being tested behaves differently when called a second time, from how it behaves when first called. You need to first test the function with one mock object, then change this mock object and test the function again with the changed mock object.
+- You need to test a function against multiple arguments quickly. For example, you need to test a function with a mock object that has a color property and then test the function again with a mock object that has a text property, but is otherwise identical to the original mock object.
```

Comment:
I'm unclear why "quickly" is a factor in the tests. It seems like the scenario is just editing the same mock object after construction.